bfsysroot/bfunwind/src/dwarf4.cpp

```diff
@@ -1476,6 +1489,137 @@ private_decode_cfi(const fd_entry &fde, register_state *state)
     return row;
 }
 
+template <typename T>
+static T* get_ptr(const void *mem, size_t off)
+{
+    return reinterpret_cast<T *>((uintptr_t)mem + off);
+}
+
+static int
+private_get_abbrev(void *debug_abbrev,
+                   uint64_t num,
+                   uint64_t *tag,
+                   bool *children)
+{
+    char *_debug_abbrev = reinterpret_cast<char *>(debug_abbrev);
+    uint64_t _abbrev;
+    uint64_t _tag;
+    uint64_t _size;
+    bool     _children;
+
+    if (num == 0) {
+        return 0;
+    }
+
+    while (true) {
+        _abbrev = dwarf4::decode_uleb128(&_debug_abbrev);
+        _tag = dwarf4::decode_uleb128(&_debug_abbrev);
+        _children = *_debug_abbrev == DW_CHILDREN_yes ? true : false;
+        _debug_abbrev += 1;
+
+        if (_abbrev == num) {
+            *tag = _tag;
+            *children = _children;
+            return 1;
+        }
+
+        // parse attributes
+        while (true) {
+            uint64_t _name;
+            uint64_t _form;
+            _name = dwarf4::decode_uleb128(&_debug_abbrev);
+            _form = dwarf4::decode_uleb128(&_debug_abbrev);
+
+            if (_name == 0 && _form == 0) {
+                break;
+            }
+        }
+    }
+
+    return 0;
+}
+
+static int
+private_get_attr(void *debug_abbrev,
+                 uint64_t abbrev,
+                 int num,
+                 uint64_t *name,
+                 uint64_t *form)
+{
+    char *_debug_abbrev = reinterpret_cast<char *>(debug_abbrev);
+    uint64_t _abbrev;
+    uint64_t _tag;
+    uint64_t _size;
+    bool     _children;
+
+    if (abbrev == 0) {
+        return 0;
+    }
+
+    while (true) {
+        uint64_t j = 0;
+        _abbrev = dwarf4::decode_uleb128(&_debug_abbrev);
+        _tag = dwarf4::decode_uleb128(&_debug_abbrev);
+        _children = *_debug_abbrev == DW_CHILDREN_yes ? true : false;
```

Comment:
these constants are defined with this case setting in the dwarf standard, similar to the `DW_OP_*`, `DW_CFA_*`, `DW_TAG_*`, etc so i'd rather keep `DW_CHILDREN_yes`